app/assets/javascripts/the_bridge_ng/controllers/marketplace_ctrl.js.coffee

```diff
@@ -30,5 +31,24 @@
       return product.integrated
     else
       return true
+
+  $scope.changeVote = (flag, product_id) ->
+    html = $('#num_'+product_id).html()
+    count = parseInt(html)
+    if flag == 'up'
+      $('#num_'+product_id).html count + 1
+      $('.up_'+product_id).hide()
+      $('.down_'+product_id).show()
+    else
+      $('#num_'+product_id).html count - 1
+      $('.up_'+product_id).show()
+      $('.down_'+product_id).hide()
+    $http.get('/vote/'+product_id)
```

Comment:
If you use `product` as the argument for this function you could do the following:

```coffeescript
$scope.changeVoteFor = (product) ->
  $http.get("/products/#{product.id}/vote")
    .success (data) -> product = data.product
```

By using the server to synchronise both the votes state (either is voted or not, and the vote count), we can get the entire product view to update itself.